src/terms/local.jl

```diff
@@ -71,19 +71,35 @@ function (external::ExternalFromFourier)(basis::PlaneWaveBasis{T}) where {T}
 end
 
 # Returns the form factors at unique values of |G + q| (in Cartesian coordinates).
-# Uses a hash map for O(1) lookup.
-function atomic_local_form_factors(basis::PlaneWaveBasis{T}, Gqs_cart::AbstractArray) where{T}
-    form_factors = IdDict{Tuple{Int,T},T}()  # IdDict for Dual compatibility
-    for G in Gqs_cart
+# Additionally, returns a mapping from any G index to a unique |G + q| index
+function atomic_local_form_factors(basis::PlaneWaveBasis{T}; q=zero(Vec3{T})) where{T}
+    Gqs_cart = [basis.model.recip_lattice * (G + q) for G in to_cpu(G_vectors(basis))]
+
+    iG2inorm_cpu = zeros(Int, length(Gqs_cart))
```

Comment:
Why not `iG2iform_cpu`, essentially that's the whole point, no ? To get a mapping from G to the correct form factor value ?
Or do you need anywhere the property that it's a mapping from G to unique G+q ?

Reply:
No, it's always tied to the form factor. `iG2iform_cpu` sounds good